Browser_IDE/executionEnvironment_Page.js

```diff
@@ -80,34 +91,52 @@ document.getElementById("canvas").addEventListener("click", async function () {
 
 // Convenience function for reporting errors, printing them to the terminal
 // and also sending a message to the main window.
-function ReportError(block, message, line, formatted=false){
+function ReportError(block, message, line, stacktrace ,formatted=false){
     let outputMessage = message != "";
+    let stackTrace = stacktrace;
+
+    // Ensure block and message are strings
+    block = block || "";
+    message = message || "";
 
-    if (outputMessage && line != null)
-        message = "Error on line "+line+": "+message;
+    // Escape only the user-provided input
+    let escapedBlock = escapeHtml(block);
+    let escapedMessage = escapeHtml(message);
 
-    if (block != null && block != "" && block != "__USERCODE__null") {
-        if (!block.startsWith(userCodeBlockIdentifier)){
-            message = "Please file a bug report and send us the following info!\n    Error in file: "+block+"\n    "+message;
-            block = "Internal Error";
+    if (outputMessage && line != null && !formatted)
+        escapedMessage = "Error on line "+line+": "+escapedMessage;
+
+    if (escapedBlock != null && escapedBlock != "" && escapedBlock != "__USERCODE__null") {
+        if (!escapedBlock.startsWith(userCodeBlockIdentifier)){
+            escapedMessage = "Please file a bug report and send us the following info!\n    Error in file: "+escapedBlock+"\n    "+escapedMessage;
+            escapedBlock = "Internal Error";
         }
         else{
-            block = block.slice(userCodeBlockIdentifier.length);
+            escapedBlock = escapedBlock.slice(userCodeBlockIdentifier.length);
         }
         if (outputMessage)
-            message = "(" + block + ") " + message;
+            escapedMessage = "(" + escapedBlock + ") " + escapedMessage;
+    }
+    
+    // Check if the stackTrace is empty
+    if (stackTrace.trim() !== "") {
+        // Format the stack trace with <details> and <summary> tags
+        stackTrace = '<pre>' + stackTrace + '</pre>';
+
+        if (outputMessage && !formatted) {
+            // If formatted is true, do not add the color: red styling
+            escapedMessage = '<summary' + (formatted ? '' : ' style="color: red;"') + '>' + escapedMessage + '</summary>';
+            escapedMessage = '<details>' + escapedMessage + stackTrace + '</details>';
+        }
     }
-
-    if (outputMessage && !formatted)
-        message = "\x1b[0m\x1b[31m" + message + "\x1b[0m";

```

Comment:
Would be good to put these lines back, since currently if there is no stack trace (like when we make a syntax error, e.g `let;`, then it shows up white still. Here's my suggestion - just make it an `else` block:
```suggestion
    else if (outputMessage && !formatted)
        escapedMessage = "\x1b[0m\x1b[31m" + escapedMessage + "\x1b[0m";
```